ironfish/src/indexers/minedBlocksIndexer.ts

```diff
@@ -215,16 +199,36 @@ export class MinedBlocksIndexer {
   }
 
   async load(): Promise<void> {
-    const headHash = await this.meta.get('headHash')
-    this.chainProcessor.hash = headHash ? Buffer.from(headHash, 'hex') : null
+    const meta = await this.loadMinedBlocksMeta()
+    this.chainProcessor.hash = meta.headHash ? Buffer.from(meta.headHash, 'hex') : null
+    if (meta.accountToRemove !== null) {
+      this.removalInProgress = true
+    }
+  }
+
+  async loadMinedBlocksMeta(): Promise<MinedBlocksDBMeta> {
+    const meta = { ...getMinedBlocksDBMetaDefaults() }
+
+    for await (const [key, value] of this.meta.getAllIter()) {
+      meta[key] = value
+    }
+
+    return meta
   }
 
-  start(): void {
+  async start(): Promise<void> {
     if (this.isStarted) {
       return
     }
     this.isStarted = true
 
+    if (this.removalInProgress) {
+      const accountName = await this.meta.get('accountToRemove')
+      if (accountName) {
+        await this.removeMinedBlocks(accountName)
+      }
+    }
```

Comment:
Code architecture wise, all this should be inside the event loop to avoid updating the head, and handling an account removal.